content/docs/source-git/_index.md

```diff
@@ -39,12 +40,13 @@ such a thing:
 
 ### Spec file
 
-I'm going to answer the elephant in the room right away: "Do we really need to
-have a spec file in our upstream repo or a source-git repo?!" — No, you don't,
-though storing the spec file directly in the repository is the most convenient.
-Alternatively, packit needs to have a way to obtain the spec file, so, you can:
+We are going to answer the elephant in the room right away: "Do we really need to
+[have a spec file in our upstream repo or a source-git repo?!]({{< ref "faq#how-can-i-download-rpm-spec-file-if-it-is-not-part-of-upstream-repository" >}})"
+No, you don't, though storing the spec file directly in the repository
+will save you a lot of trouble. Alternatively, packit needs to have a way to
```

Comment:
What "trouble"? :)

Storing the spec file in the same tree as the source code creates a tight-coupling between the two. This might be something that is wanted or something that is not.

If the spec file is stored next to the source code and it's used for building the package, each build can be traced back to a given commit and so a given (source code, spec file) pair. It is obvious how to find the version of the spec file which was used.

If the spec file is stored some other place, the spec file can be changed without this change being reflected in the history of the source code. So tracing back a build to a given (source code, spec file) pair becomes problematic or would require additional tooling. Maybe this is the trouble you though of? :slightly_smiling_face:

Reply:
Trouble = someone can change the spec file in the other place and suddenly builds would start failing while one would need to go and fix the spec which sits in a different repo. To me this feels like storing setup.py & setup.cfg in a different repo...

> If the spec file is stored some other place, the spec file can be changed without this change being reflected in the history of the source code. So tracing back a build to a given (source code, spec file) pair becomes problematic or would require additional tooling. Maybe this is the trouble you though of? slightly_smiling_face

exactly, that's why I don't understand the hatred for "spec file in upstream" scenario

Reply:
I see. Then I think this doc should explain these considerations. "Trouble" might not be self-explanatory enough to all readers :)

In general, my point is, that we should try to avoid using language that might be confusing to readers who didn't spend this much time studying the topic and we should go out of our way explaining the recommendations and decisions we took along the way. Writing "will save you a lot of trouble" doesn't meet any of this criteria.

Reply:
I'll elaborate, good point.